### PR TITLE
Enable go modules

### PR DESCRIPTION
**Pull request by stevesloka**
Updates #57 

This PR is spread out over a few commits to accomplish the following, my thought was to get these bits integrated first then move to CI. 

I can break this up into multiple PR's if that's easier:

1. Switching from Glide to go modules
2. Fixes up Makefile for modules
3. Updates all references to `lyft/ratelimit` --> `envoyproxy/ratelimit`
4. Updates docker-compose & dockerfile

### CODE REVIEW

**Comment by stevesloka** [comment]
I ended up bumping the travis file as well to make sure the build still succeeded. Follow up PR will enable pushing an image to a repo.

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 sudo:     required
 language: go
-go:       "1.11"
+go:       "1.14"
 services: redis-server
 before_install: sudo apt-get update -y && sudo apt-get install stunnel4 -y
-install: make bootstrap bootstrap_redis_tls
+install: make bootstrap_redis_tls
```
Comment (junr03) [inline .travis.yml]: why is bootstrap gone here, but still present in the Makefile? Should mockgen install remain part of bootstrap tests?

File: Dockerfile [hunk 2]
```diff
@@ -1,18 +1,16 @@
-FROM golang:1.10.4 AS build
-WORKDIR /go/src/github.com/lyft/ratelimit
+FROM golang:1.14 AS build
+WORKDIR /ratelimit
+
+ENV GOPROXY=https://proxy.golang.org
```
Comment (junr03) [inline Dockerfile]: sorry, not terribly familiar with go modules. Why is this needed in docker but not in the makefile for local builds?

File: Makefile [hunk 3]
```diff
@@ -1,18 +1,13 @@
-ifeq ("$(GOPATH)","")
-$(error GOPATH must be set)
-endif
+export GO111MODULE=on
+MODULE = github.com/envoyproxy/ratelimit
 
 SHELL := /bin/bash
-GOREPO := ${GOPATH}/src/github.com/lyft/ratelimit
+GOREPO := ${GOPATH}/src/github.com/envoyproxy/ratelimit
```
Comment (junr03) [inline Makefile]: looks like this is not used anymore in the rest of the file. Not sure if this is needed for the internal Lyft build.

File: .travis.yml (hunk 1 above)
Comment (stevesloka) [reply .travis.yml]: Thanks for catching, at one point I removed bootstrap entirely, then put it back. I'm going to keep it there so I will restore to the travis file.

File: Dockerfile (hunk 2 above)
Comment (stevesloka) [reply Dockerfile]: It's a mirror that can be used to speed up module downloads. We use it in Contour.

File: Makefile (hunk 3 above)
Comment (stevesloka) [reply Makefile]: This isn't used anywhere else in the makefile we can delete it.

**Comment by stevesloka** [comment]
I fixed up your review comments @junr03, thanks!

**Comment by ezgitek** [comment]
Hi, thanks for the changes! Any ETA on when this will be merged? It's breaking our Docker builds after the re-org.

```
Step 8/13 : RUN CGO_ENABLED=0 GOOS=linux go build -o /usr/local/bin/ratelimit -ldflags="-w -s" -v github.com/envoyproxy/ratelimit/src/service_cmd
 ---> Running in a2a16942a787
src/service_cmd/main.go:3:8: cannot find package "github.com/lyft/ratelimit/src/service_cmd/runner" in any of:
	/go/src/github.com/envoyproxy/ratelimit/vendor/github.com/lyft/ratelimit/src/service_cmd/runner (vendor tree)
	/usr/local/go/src/github.com/lyft/ratelimit/src/service_cmd/runner (from $GOROOT)
	/go/src/github.com/lyft/ratelimit/src/service_cmd/runner (from $GOPATH)
The command '/bin/sh -c CGO_ENABLED=0 GOOS=linux go build -o /usr/local/bin/ratelimit -ldflags="-w -s" -v github.com/envoyproxy/ratelimit/src/service_cmd' returned a non-zero code: 1
time="2020-03-09T10:36:20Z" level=fatal msg="exit status 1"
```
